site/src/pages/api/blocks.api.ts

```diff
@@ -105,6 +106,21 @@ export default createApiKeyRequiredHandler<null, ApiSearchResponse>()
         .map(([block, _]) => block);
     }
 
+    for (const block of data) {
+      // Generate absolute URLs for block icons
+      if (block.icon && !block.icon.startsWith("http")) {
+        block.icon = `${FRONTEND_URL}${block.icon}`;
+      }
+
+      for (const variant of block.variants ?? []) {
+        if (variant.icon.startsWith("public")) {
```

Comment:
This one's a little more tricky. I don't think we have a standard for how the block-variant URLs should be formatted yet. For now, all non-absolute variant icons start with `public/`, hence the condition here.

Reply:
They are formatted like normal URLs - absolute ones will start with a protocol.